qiskit_ibm_runtime/session.py

```diff
@@ -279,28 +280,28 @@ def service(self) -> QiskitRuntimeService:
         return self._service
 
     @classmethod
-    def from_id(
-        cls,
-        session_id: str,
-        service: Optional[QiskitRuntimeService] = None,
-        backend: Optional[Union[str, IBMBackend]] = None,
-    ) -> "Session":
+    def from_id(cls, session_id: str, service: QiskitRuntimeService) -> "Session":
```

Comment:
You need to deprecate the parameter before removing it. So during the deprecation period it needs to check if the backends match.